src/compiler/emitter.ts

```diff
@@ -1257,26 +1257,34 @@ var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, ge
 
                     // Children
                     if (children) {
-                        for (let i = 0; i < children.length; i++) {
-                            // Don't emit empty expressions
-                            if (children[i].kind === SyntaxKind.JsxExpression && !((<JsxExpression>children[i]).expression)) {
-                                continue;
+                        const isChildEmittable = (child: JsxChild): boolean => {

```

Comment:
@DanielRosenwasser why can not we do this as part of the original loop, why do we need a separate filter pass?